rewrite-java/src/main/java/org/openrewrite/java/service/AnnotationService.java

```diff
@@ -53,7 +52,55 @@ public List<J.Annotation> getAllAnnotations(Cursor cursor) {
             return ((J.TypeParameters) j).getAnnotations();
         } else if (j instanceof J.Package) {
             return ((J.Package) j).getAnnotations();
+        } else if (j instanceof J.AnnotatedType) {
+            return getAllAnnotations((J.AnnotatedType) j);
+        } else if (j instanceof J.ArrayType) {
+            return getAllAnnotations((J.ArrayType) j);
+        } else if (j instanceof J.FieldAccess) {
+            return getAllAnnotations((J.FieldAccess) j);
+        } else if (j instanceof J.Identifier) {
+            return getAllAnnotations((J.Identifier) j);
+        }
+        return emptyList();
+    }
+
+    private List<J.Annotation> getAllAnnotations(J j) {
+        if (j instanceof J.AnnotatedType) {
+            return getAllAnnotations((J.AnnotatedType) j);
+        } else if (j instanceof J.ArrayType) {
+            return getAllAnnotations((J.ArrayType) j);
+        } else if (j instanceof J.Identifier) {
+            return getAllAnnotations((J.Identifier) j);
+        } else if (j instanceof J.FieldAccess) {
+            return getAllAnnotations((J.FieldAccess) j);
         }
         return emptyList();
     }
+
+    private List<J.Annotation> getAllAnnotations(J.AnnotatedType annotatedType) {
+        List<J.Annotation> annotations = new ArrayList<>(annotatedType.getAnnotations().size());
+        annotations.addAll(annotatedType.getAnnotations());
+        annotations.addAll(getAllAnnotations(annotatedType.getTypeExpression()));
+        return annotations;
+    }
+
+    private List<J.Annotation> getAllAnnotations(J.ArrayType arrayType) {
+        List<J.Annotation> annotations = new ArrayList<>(arrayType.getAnnotations() == null ? 0 : arrayType.getAnnotations().size());
+        annotations.addAll(getAllAnnotations(arrayType.getElementType()));
+        if (arrayType.getAnnotations() != null) {
+            annotations.addAll(arrayType.getAnnotations());
+        }
+        return annotations;
+    }
+
+    private List<J.Annotation> getAllAnnotations(J.FieldAccess fieldAccess) {
+        List<J.Annotation> annotations = new ArrayList<>();
+        annotations.addAll(getAllAnnotations(fieldAccess.getTarget()));
```

Comment:
This I am a bit unsure about. But AFAIK the type annotations can only occur on the name and not the target.